### PR TITLE
Bipartite network api update

### PR DESCRIPTION
**Pull request by d-callan**
resolves https://github.com/VEuPathDB/web-monorepo/issues/764

this will keep from breaking things that are currently working. idk if there were other changes you were anticipating..

### CODE REVIEW

**Comment by d-callan** [comment]
side note.. seems the bipartitenetwork viz is maybe more properly called a correlationbipartitenetwork ?? not urgent obviously, just a thought i had while looking at this

**Review by asizemore — COMMENTED** [review]
I think it's important we separate the data we use to make the network from the drawing of the network. This concept will become more crucial as we get more data for nodes and links.
Currently there is a `BipartiteNetworkData` type defined in both the `components` and `eda` libraries. They are currently kept in sync (there's a long discussion about this in another pr if you're interested). But the point is that I think we should un-sync them. The bipartite data coming from the backend should remain as `BipartiteNetworkData` with the api changes, and then the `components` type should be the actual network, not the data. So it should say "strokeWidth" instead of `weight`, for example. I want to call that type `BipartiteNetwork` but that could be confusing with the component being named the same thing. @dmfalke might have an opinion here.

At the very least, if we do not decide to go the above route, i think there are some more `strokeWidth` instances in the `components` library that should be updated. For example the `Link` component. I would guess when you tested the changes all the links were drawn with the same stroke width?

File: packages/libs/eda/src/lib/core/components/visualizations/implementations/BipartiteNetworkVisualization.tsx [hunk 1]
```diff
@@ -285,12 +287,11 @@ function BipartiteNetworkViz(props: VisualizationProps<Options>) {
   const lineLegendItems: LegendItemsProps[] = [
     ...Array(nLineItemsInLegend).keys(),
   ].map((i) => {
-    const adjustedStrokeWidth =
-      maxDataStrokeWidth -
-      ((maxDataStrokeWidth - minDataStrokeWidth) / (nLineItemsInLegend - 1)) *
-        i;
+    const weightLabel =
```
[the remark this answers is not in the document]
Comment (d-callan) [reply packages/libs/eda/src/lib/core/components/visualizations/implementations/BipartiteNetworkVisualization.tsx:290]: I thought these were the labels for the weights. Meaning even if the weights get rescaled to make them visible or something the legend labels would still reflect the underlying weights..

File: packages/libs/eda/src/lib/core/components/visualizations/implementations/BipartiteNetworkVisualization.tsx [hunk 2]
```diff
@@ -162,21 +164,21 @@ function BipartiteNetworkViz(props: VisualizationProps<Options>) {
...
+  const maxDataWeight = Math.max(...dataWeights);
 
   // Clean and finalize data format. Specifically, assign link colors, add display labels
   const cleanedData = useMemo(() => {
     if (!data.value) return undefined;
 
     // Create map that will adjust each link's stroke width so that all link stroke widths span an appropriate range for this viz.
-    const strokeWidthMap = scaleLinear()
-      .domain([minDataStrokeWidth, maxDataStrokeWidth])
+    const weightMap = scaleLinear()
```
[the remark this answers is not in the document]
Comment (d-callan) [reply packages/libs/eda/src/lib/core/components/visualizations/implementations/BipartiteNetworkVisualization.tsx:180]: I went back and forth here a couple times before landing on weight, but I think I like the idea of saying weightToStrokeMap or similar

**Comment by d-callan** [comment]
I'd be curious to read through the other discussion. And I have questions about the stroke vs weight thing for the component, but I'll save them for Mon probably

File: packages/libs/eda/src/lib/core/components/visualizations/implementations/BipartiteNetworkVisualization.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (asizemore) [reply packages/libs/eda/src/lib/core/components/visualizations/implementations/BipartiteNetworkVisualization.tsx:290]: I see your point, that makes sense

**Comment by d-callan** [comment]
ok. so i changed some names to separate the viz and component types as requested. i think given how things are, its reasonable. though i do wonder if were potentially backing ourselves into a corner. are we very confident the component will never need to know about the underlying data? that will always be either the backend or the vizs job? given the way weve engineered things so far im ok w continuing w that assumption, just made me a bit nervous.

**Comment by asizemore** [comment]
Yes i think from a philosophical standpoint the component should not know about the data. It should be only drawing a network, exactly as given. It's everyone else's job to provide the context, and let the component be totally independent. There might be something that breaks that idea someday, but for now this setup seems like it should best handle all of our use cases.

**Review by asizemore — APPROVED** [review]

**Comment by dmfalke** [comment]
> I want to call that type `BipartiteNetwork` but that could be confusing with the component being named the same thing. @dmfalke might have an opinion here.

I'm not sure which type you want to call `BipartiteNetwork`. Regardless, my instinct is to say that we should not have a component named `BipartiteNetwork`, and instead should be something like `BipartiteNetworkPlot`. My thinking is that the component represents one way to "visualize" the network data. Does that make sense?

**Comment by asizemore** [comment]
@dmfalke sounds good. I added this renaming task to the bipartite network polishing gh issue.

**Comment by dmfalke** [comment]
Great. I wonder if there are other components that should have `Plot` appended to their name. Would you mind adding a note to check in the polish-up issue?

**Pull request merged**